node_modules/oae-authentication/lib/util.js

```diff
@@ -152,6 +152,15 @@ var logAuthenticationSuccess = module.exports.logAuthenticationSuccess = functio
         data.imposterId = imposter.id;
     }
 
+Cassandra.runQuery('UPDATE "Principals" SET "lastConnected" = ? WHERE "principalId" = ?', [LastConnected.toString() , user.id], function(err) {
+        if (err) {
+            log().error({'err': err}, 'Error changing a LastConnected');
+           
+        }
+
+        
+    });
+
     log().info(data, util.format('Login for "%s" to tenant "%s" from "%s"', user.id, tenantAlias, req.headers['x-forwarded-for']));
```

Comment:
This function has a single purpose which is outlined in the JsDoc: `Outputs a log message for a successful authentication` so this update must be moved somewhere else after the authentication was successfully completed.

Also, take into consideration that this is a `cql` call and all direct interactions with the database should go into the "Data Access Object" file in `oae-principals/lib/internal/dao.js` so to keep thins tidy and organised.